cli/resolver/tag_test.go

```diff
@@ -0,0 +1,207 @@
+package resolver
+
+import (
+	"net/url"
+	"sort"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/client"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTagResolverQueryParams(t *testing.T) {
+	testCases := []struct {
+		Name            string
+		EntityType      string
+		Target          string
+		ExpectedQueries []url.Values
+	}{
+		{
+			Name:       "unqualified deploy",
+			EntityType: "deploy",
+			Target:     "d*",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType: []string{"deploy"},
+				client.TagQueryParamFuzz: []string{"d*"},
+			}},
+		},
+		{
+			Name:       "fully-qualified deploy",
+			EntityType: "deploy",
+			Target:     "d:1",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType:    []string{"deploy"},
+				client.TagQueryParamFuzz:    []string{"d"},
+				client.TagQueryParamVersion: []string{"1"},
+			}},
+		},
+		{
+			Name:       "environment",
+			EntityType: "environment",
+			Target:     "e*",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType: []string{"environment"},
+				client.TagQueryParamFuzz: []string{"e*"},
+			}},
+		},
+		{
+			Name:       "job",
+			EntityType: "job",
+			Target:     "j*",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType: []string{"job"},
+				client.TagQueryParamFuzz: []string{"j*"},
+			}},
+		},
+		{
+			Name:       "unqualified load_balancer",
+			EntityType: "load_balancer",
+			Target:     "l*",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType: []string{"load_balancer"},
+				client.TagQueryParamFuzz: []string{"l*"},
+			}},
+		},
+		{
+			Name:       "fully-qualified load_balancer",
+			EntityType: "load_balancer",
+			Target:     "e*:l*",
+			ExpectedQueries: []url.Values{
+				{
+					client.TagQueryParamType: []string{"environment"},
+					client.TagQueryParamFuzz: []string{"e*"},
+				},
+				{
+					client.TagQueryParamType:          []string{"load_balancer"},
+					client.TagQueryParamFuzz:          []string{"l*"},
+					client.TagQueryParamEnvironmentID: []string{"eid"},
+				},
+			},
+		},
+		{
+			Name:       "unqualified service",
+			EntityType: "service",
+			Target:     "s*",
+			ExpectedQueries: []url.Values{{
+				client.TagQueryParamType: []string{"service"},
+				client.TagQueryParamFuzz: []string{"s*"},
+			}},
+		},
+		{
+			Name:       "fully-qualified service",
+			EntityType: "service",
+			Target:     "e*:s*",
+			ExpectedQueries: []url.Values{
+				{
+					client.TagQueryParamType: []string{"environment"},
+					client.TagQueryParamFuzz: []string{"e*"},
+				},
+				{
+					client.TagQueryParamType:          []string{"service"},
+					client.TagQueryParamFuzz:          []string{"s*"},
+					client.TagQueryParamEnvironmentID: []string{"eid"},
```

Comment:
@seshi: is this too much "magic"? Is it clear where `"eid"` comes from?

Reply:
Assuming it comes from your mock because e* will always resolve to eid? If not then yes this is too much magic.

Reply:
ok, fixed it to make it more clear